sdk/smx.ts

```diff
@@ -233,6 +235,97 @@ export class SMXStage {
+    {
+        // The 'l' command used to set lights, but it's now only used to turn lights off
+        // for cases like this.
+        string sData = "l";
+        sData.append(108, 0);
+        sData += "\n";
+        for(int iPad = 0; iPad < 2; ++iPad)
+            m_pDevices[iPad]->SendCommandLocked(sData);
+    }
+
+    m_SentPanelTestModeAtTicks = now;
+    m_LastSentPanelTestMode = m_PanelTestMode;
+    for(int iPad = 0; iPad < 2; ++iPad)
+        m_pDevices[iPad]->SendCommandLocked(ssprintf("t %c\n", m_PanelTestMode));
+}
+*/
+
+  setPanelTestMode(mode: PanelTestMode) {
+    // If we want to turn panel test mode off...
+    if (mode === PanelTestMode.Off) {
+      // We don't want to send the "Off" command multiple times, so only send it if
+      // it's currently activated
+      if (this.panelTestMode !== PanelTestMode.Off) {
+        if (this.testModeIntervalHandle !== null) {
+          clearInterval(this.testModeIntervalHandle);
+          this.testModeIntervalHandle = null;
+        }
+        // Turn off panel test mode, and send "Off" event
+        this.panelTestMode = mode;
+        this.events.output$.push(Uint8Array.of(API_COMMAND.SET_PANEL_TEST_MODE, mode));
+
+        // TODO: Do we need to do anything with this? Does this even need to be called to flush
+        // the queue?
+        this.events.ackReports$.firstToPromise();
+      }
+
+      // Either we're already off, or we sent the off command, so just return.
+      return;
+    }
+
+    // We only need to run this when the current mode is not the same as the requested mode
+    if (this.panelTestMode !== mode) {
+      this.panelTestMode = mode;
+
+      /**
+       * the 'l' command used to set lights, but it's now only used to turn lights off
+       * for cases like this
+       * Lights are always updated together (for some reason??)
+       * 2 pads * 9 panels * 25 lights each * 3 (RGB) = 1350
+       * The source code uses `108` and I'm really unsure why
+       *
+       * TODO: Does this even do anything?
+       */
+      this.events.output$.push(Uint8Array.of(API_COMMAND.SET_LIGHTS_OLD, ...padData([], 1350, 0), char2byte("\n")));
+
+      // TODO: Do we need to do anything with this? Does this even need to be called to flush
+      // the queue?
+      this.events.ackReports$.firstToPromise();
+
+      // Send the Panel Test Mode command
+      this.events.output$.push(Uint8Array.of(API_COMMAND.SET_PANEL_TEST_MODE, mode));
+
+      // TODO: Do we need to do anything with this? Does this even need to be called to flush
+      // the queue?
+      this.events.ackReports$.firstToPromise();
+
+      // The Panel Test Mode command needs to be resent every second
+      this.testModeIntervalHandle = setInterval(() => {
+        this.events.output$.push(Uint8Array.of(API_COMMAND.SET_PANEL_TEST_MODE, mode));
+
+        // TODO: Do I need to call this to consume the event?
+        this.events.ackReports$.firstToPromise();
```

Comment:
I don't think you need to worry about consuming these. I think because the are acks to unattended commands we can just let them silently discarded in the background.